### ISSUE TITLE
probsToLogits uses base-10 log10 instead of natural log — corrupts Categorical/Bernoulli logProb and entropy

### ISSUE THREAD

**Issue by schnecki** [hasktorch/hasktorch#799]
## Summary

[`probsToLogits`](https://github.com/hasktorch/hasktorch/blob/master/hasktorch/src/Torch/Distributions/Distribution.hs#L57-L61) computes logits with `F.log10` (base-10 logarithm). It must use the natural logarithm. This silently corrupts `logProb` and `entropy` for both `Categorical` and `Bernoulli`.

```haskell
probsToLogits :: Bool -> D.Tensor -> D.Tensor -- isBinary=False
probsToLogits isBinary probs =
  if isBinary
    then F.log10 psClamped `F.sub` F.log1p (F.mulScalar (-1.0 :: Float) psClamped)
    else F.log10 psClamped
  where
    psClamped = clampProbs probs
```

PyTorch's reference implementation ([`torch.distributions.utils.probs_to_logits`](https://github.com/pytorch/pytorch/blob/main/torch/distributions/utils.py)) uses natural `log`:

```python
def probs_to_logits(probs, is_binary=False):
    ps_clamped = clamp_probs(probs)
    if is_binary:
        return torch.log(ps_clamped) - torch.log1p(-ps_clamped)
    return torch.log(ps_clamped)
```

Note the current code is also internally inconsistent: the `is_binary` branch mixes `log10` with `log1p` (natural), so the binary logit is neither a correct base-10 nor base-e quantity.

## Impact

`Categorical.fromProbs` and `Bernoulli.fromProbs` build `logits = probsToLogits probs`, and both `logProb` and `entropy` are derived from those logits. Consequently:

- `Categorical.logProb dist a` returns `log10(p_a) = ln(p_a) / ln(10) ≈ 0.4343 · ln(p_a)` instead of `ln(p_a)`.
- `Categorical.entropy` (`-Σ logits · probs`) returns `0.4343 · H` instead of the true entropy `H`.
- Same `0.4343` scaling for `Bernoulli`.

This breaks any code that relies on these being true log-probabilities / entropy — e.g. policy-gradient / PPO surrogate objectives, importance ratios `exp(logp_new − logp_old)`, KL estimates, and entropy bonuses. The error is self-consistent (new and old log-probs share the same scale factor), so it does **not** crash or NaN; it just scales gradients by `1/ln(10) ≈ 0.4343`, mis-scales clip thresholds, and yields wrong entropy — which is very easy to miss.

## Reproduction

```haskell
import qualified Torch.Distributions.Categorical  as C
import qualified Torch.Distributions.Distribution as D
import qualified Torch                            as T

main :: IO ()
main = do
  let probs = T.asTensor ([0.1, 0.2, 0.3, 0.4] :: [Float])
      dist  = C.fromProbs probs
  -- logProb of action 3 (p = 0.4):
  print (T.asValue (D.logProb dist (T.asTensor (3 :: Int))) :: Float)
  -- observed: -0.39794 (= log10 0.4)
  -- expected: -0.91629 (= ln    0.4)
  print (T.asValue (C.entropy dist) :: Float)
  -- observed: 0.5766  (= 0.4343 * 1.2799)
  -- expected: 1.2799  (natural-log entropy)
```

## Fix

Replace both `F.log10` with the natural log `F.log`:

```haskell
probsToLogits isBinary probs =
  if isBinary
    then F.log psClamped `F.sub` F.log1p (F.mulScalar (-1.0 :: Float) psClamped)
    else F.log psClamped
  where
    psClamped = clampProbs probs
```

This makes `probsToLogits` the exact inverse of `logitsToProbs` (softmax/sigmoid) up to clamping, matching PyTorch. A regression test asserting `logProb dist a == log (probs ! a)` for a normalized `Categorical`, and `entropy` against the closed form `-Σ p log p`, would guard against reintroduction.
